packages/react/toggle-button/src/ToggleButton.tsx

```diff
@@ -1,10 +1,199 @@
+
+const GROUP_NAME = 'ToggleButtonGroup';
+const GROUP_DEFAULT_TAG = 'div';
+const GROUP_CONTEXT_NAME = 'ToggleButtonGroupContext';
+
+type ToggleButtonGroupContextValue = {
+  value: string[];
+  setValue: React.Dispatch<React.SetStateAction<string[]>>;
+  handleChange(value: string): void;
+  name: typeof GROUP_CONTEXT_NAME;
+  rovingFocus: boolean;
+};
+
+const ToggleButtonGroupContext = React.createContext<ToggleButtonGroupContextValue | null>(null);
+ToggleButtonGroupContext.displayName = GROUP_CONTEXT_NAME;
+
+type ToggleButtonGroupSharedProps = {
+  /** Whether or not a selection in the group is required after initial selection */
+  selectionIsRequired?: boolean;
+  /** Whether or not the group should maintain roving focus of its buttons */
+  rovingFocus?: boolean;
+};
+
+type ToggleButtonGroupOwnProps = ToggleButtonGroupSharedProps & {
+  /** The controlled value of the toggled button */
+  value?: string[];
+  /** The uncontrolled value of the toggled button */
+  defaultValue?: string[];
+  /** A function called when the value of the toggled buttons changes */
+  onValueChange?(value: string[]): void;
+};
+
+const ToggleButtonGroup = forwardRefWithAs<typeof GROUP_DEFAULT_TAG, ToggleButtonGroupOwnProps>(
+  (props, forwardedRef) => {
+    const {
+      as: Comp = GROUP_DEFAULT_TAG,
+      value: valueProp,
+      defaultValue,
+      onValueChange,
+      children,
+      selectionIsRequired = false,
+      rovingFocus = false,
+      ...groupProps
+    } = props;
+
+    const [value, setValue] = useControlledState<string[]>({
+      prop: valueProp,
+      defaultProp: defaultValue || [],
+      onChange: onValueChange,
+      isEqual: stringArraysAreEqual,
+    });
+
+    function handleChange(buttonValue: string) {
+      if (!value || value.length < 1) {
+        return setValue([buttonValue]);
+      }
+
+      if (selectionIsRequired && value.length === 1 && value.includes(buttonValue)) {
+        return;
+      }
+
+      setValue(
+        value.includes(buttonValue)
+          ? value.filter((v) => v !== buttonValue)
+          : value.concat(buttonValue).sort()
+      );
+    }
+
+    const context: ToggleButtonGroupContextValue = {
+      setValue,
+      value,
+      handleChange,
+      rovingFocus,
+      name: GROUP_CONTEXT_NAME,
+    };
+
+    return (
+      <Comp {...getPartDataAttrObj(GROUP_NAME)} role="group" ref={forwardedRef} {...groupProps}>
+        <ToggleButtonGroupContext.Provider value={context}>
+          <ToggleButtonGroupInner rovingFocus={rovingFocus}>{children}</ToggleButtonGroupInner>
+        </ToggleButtonGroupContext.Provider>
+      </Comp>
+    );
+  }
+);
+
+function ToggleButtonGroupInner(props: { rovingFocus: boolean; children: React.ReactNode }) {
+  return props.rovingFocus ? (
+    <RovingFocusGroup loop>{props.children}</RovingFocusGroup>
+  ) : (
+    <React.Fragment>{props.children}</React.Fragment>
+  );
+}
```

Comment:
I'm struggling to work out why this needs a separate `ToggleButtonGroupInner` component. Can we move it inline please? 🙏 

I can see you're re-using it in the exclusive one but I'm not convinced that a ternary warrants a whole new component/abstraction and the boilerplate that comes with it. Feels a bit like DRY for DRY sake.

```jsx
    return (
      <Comp {...getPartDataAttrObj(GROUP_NAME)} role="group" ref={forwardedRef} {...groupProps}>
        <ToggleButtonGroupContext.Provider value={context}>
          {rovingFocus ? <RovingFocusGroup loop>{children}</RovingFocusGroup> : children}
        </ToggleButtonGroupContext.Provider>
      </Comp>
    );
```